packages/indicators/src/builders/buildArithmetic.ts

```diff
@@ -102,18 +125,48 @@ const buildAnalyticValues = (analyticClusters: AnalyticCluster[], formula: strin
     }))
     .filter(({ value }) => isFinite(value));
 
+const getAggregationTypesByCode = (config: ArithmeticConfig): Record<string, string | string[]> => {
+  const { formula, aggregation, parameters = [] } = config;
+
+  if (typeof aggregation === 'object') {
+    return aggregation;
+  }
+
+  const parameterCodes = parameters.map(p => p.code);
+  const elementCodes = getVariables(formula).filter(code => !parameterCodes.includes(code));
+  return Object.fromEntries(elementCodes.map(code => [code, aggregation]));
+};
+
+const getAggregationsByCode = (config: ArithmeticConfig) => {
+  const aggregationTypesByCode = getAggregationTypesByCode(config);
+
+  return Object.fromEntries(
+    Object.entries(aggregationTypesByCode).map(([code, aggregationTypes]) => {
+      const aggregations = toArray(aggregationTypes).map(type => ({ type }));
+      return [code, aggregations];
+    }),
+  );
+};
+
 const fetchAnalyticsAndElements = async (
   api: IndicatorApiInterface,
   config: ArithmeticConfig,
   fetchOptions: FetchOptions,
 ) => {
-  const { formula, aggregation: aggregationSpecs } = config;
+  const { parameters = [] } = config;
+
   const aggregator = api.getAggregator();
-  const aggregationsByCode = getAggregationsByCode(aggregationSpecs, formula);
-  const analytics = await fetchAnalytics(aggregator, aggregationsByCode, fetchOptions);
-  const dataElements = Object.keys(aggregationsByCode);
+  const aggregationsByCode = getAggregationsByCode(config);
+  const formulaAnalytics = await fetchAnalytics(aggregator, aggregationsByCode, fetchOptions);
+  const formulaElements = Object.keys(aggregationsByCode);
 
-  return { analytics, dataElements };
+  const parameterAnalytics = await api.buildAnalyticsForIndicators(parameters, fetchOptions);
+  const parameterElements = parameters.map(p => p.code);
+
+  return {
+    analytics: formulaAnalytics.concat(parameterAnalytics),
+    dataElements: getUniqueEntries(formulaElements.concat(parameterElements)),
```

Comment:
Why would there be the same code in formulaElements and parameterElements? (i.e. why the `getUniqueEntries`)

Reply:
By accident 😛 Parameters would then take precedence over formula elements. Unique entries are used here mainly because  an element may appear more than once in a formula, so it's just about keeping our data structures clean (it's optional in practice)

Reply:
By the way I haven't added a test case to assert the above behaviour, since I thought we could be classified as "unpredictable" (not part of the indicators' contract)

Reply:
Yep, happy with that!